functions/models/src/helpers/schemaConverter.ts

```diff
@@ -7,13 +7,13 @@
 //
 
 import { type z } from 'zod'
+import { type Lazy } from './lazy'
 
-/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-export class SchemaConverter<Schema extends z.ZodType<any, any, any>> {
+export class SchemaConverter<Schema extends z.ZodTypeAny, Encoded> {
```

Comment:
As a consumer of engage-package,  `Encode` allows to get raw Firestore data interface. There are two types we need to differentiate. 


* result of zod cast - `z.output<Schema>` - this is the type that is used in the firebase application
* result of `encode`, which should bring back Firestore-compliant object

`Encode` represents what's stored in Firestore, so it cannot be directly inferred from `Schema` in this scenario. `Schema` might parse, modify, alternate structure, `Encode` must return raw Firestore data. 


Example:

```
const converter =  new SchemaConverter({
  schema: z.object({ title: optionalish(z.string()) }),
  encode: (object) => ({ title: object.title ?? null })
})
```

`z.input` is `{ title?: string | null | undefined }`
`z.output` is `{ title?: string | undefined }` 
`Encoded` is `{ title: string | null }`